File: Web/heatmap/src/main/java/com/hs/heatmap/service/ReportService.java
package com.hs.heatmap.service;

import com.hs.heatmap.model.Report;
import com.hs.heatmap.model.ReportAgeGender;

import java.util.List;

public interface ReportService {

    List<Report> getAllReport();

    List<Report> getReportByDate(String date, int cameraID);

    List<Report> getReportByMonth(String month, int cameraID);

    List getReportAreaByMonth(String month, int areaID);

    List getReportStoreByMonth(String month, int storeID);

    List getReportAgeGenderStoreByMonth(String month, int storeID);

    List<Report> getReportByTime(String date, int cameraID, String timeFrom, String timeTo);

    List<ReportAgeGender> getReportAgeGenderByTime(String date, int cameraID, String timeFrom, String timeTo);

    List getReportAreaByTime(String date, int areaID, String timeFrom, String timeTo);

    List getReportAgeGenderAreaByTime(String date, int areaID, String timeFrom, String timeTo);

    List getReportStoreByTime(String date, int storeID, String timeFrom, String timeTo);

    List<Report> getHeatmapByDate(String date, int cameraID);

    Report createReport(Report report);

    List getReportAgeGenderStoreByTime(String selectedDate, int storeID, String timeFrom, String timeTo);
}

File: Web/v2.0 - Change Area to Zone/WebApp/src/app/services/report.service.ts
import { Injectable } from '@angular/core';
import { Observable, BehaviorSubject } from 'rxjs';
import { Report } from '../models/report.model';
import { HttpClient } from '@angular/common/http';
import { Camera } from '../models/camera.model';

@Injectable({
  providedIn: 'root'
})
export class ReportService {

  // private messageSource = new BehaviorSubject('default message');
  // currentMessage = this.messageSource.asObservable();


  private reportUrl = 'http://localhost:8080/api/report/';
  private headers = {
    'Content-Type': 'application/json',
    'Access-Control-Allow-Origin': '*',
    'Access-Control-Allow-Methods': 'GET, POST, OPTIONS, PUT, PATCH, DELETE'
  };

  constructor(private http: HttpClient) { }

  getReportByDate(selectedDate: String, cameraID: number): Observable<Report[]> {
    return this.http.get<Report[]>(this.reportUrl + 'getReportByDate/' + selectedDate + '/' + cameraID);
  }

  getReportByTime(selectedDate: String, cameraID: number, timeFrom: String, timeTo: String): Observable<Report[]> {
    return this.http.get<Report[]>(this.reportUrl + 'getReportByTime/' + selectedDate + '/'
                                                                       + cameraID + '/'
                                                                       + timeFrom + '/'
                                                                       + timeTo + '/');
  }

  getReportZoneByTime(selectedDate: String, araID: number, timeFrom: String, timeTo: String): Observable<any[]> {
    return this.http.get<any[]>(this.reportUrl + 'getReportZoneByTime/' + selectedDate + '/'
                                                                       + araID + '/'
                                                                       + timeFrom + '/'
                                                                       + timeTo);
  }

  getReportStoreByTime(selectedDate: String, storeID: number, timeFrom: String, timeTo: String): Observable<any[]> {
    return this.http.get<any[]>(this.reportUrl + 'getReportStoreByTime/' + selectedDate + '/'
                                                                       + storeID + '/'
                                                                       + timeFrom + '/'
                                                                       + timeTo);
  }

  getReportAgeGenderStoreByTime(selectedDate: String, storeID: number, timeFrom: String, timeTo: String): Observable<any[]> {
    return this.http.get<any[]>(this.reportUrl + 'getReportAgeGenderStoreByTime/' + selectedDate + '/'
                                                                       + storeID + '/'
                                                                       + timeFrom + '/'
                                                                       + timeTo);
  }

  getReportByMonth(selectedMonth: String, cameraID: number): Observable<Report[]> {
    return this.http.get<Report[]>(this.reportUrl + 'getReportByMonth/' + selectedMonth + '/' + cameraID);
  }

  getHeatmapListByDate(selectedDate: String, cameraID: number): Observable<Report[]> {
    return this.http.get<Report[]>(this.reportUrl + 'getHeatmapByDate/' + selectedDate + '/' + cameraID);
  }

  getReportZoneByMonth(selectedMonth: String, zoneID: number): Observable<any[]> {
    return this.http.get<any[]>(this.reportUrl + 'getReportZoneByMonth/' + selectedMonth + '/' + zoneID);
  }

  getReportStoreByMonth(selectedMonth: String, storeID: number): Observable<any[]> {
    return this.http.get<any[]>(this.reportUrl + 'getReportStoreByMonth/' + selectedMonth + '/' + storeID);
  }

  getReportAgeGenderStoreByMonth(selectedMonth: String, storeID: number): Observable<any[]> {
    return this.http.get<any[]>(this.reportUrl + 'getReportAgeGenderStoreByMonth/' + selectedMonth + '/' + storeID);
  }

}
